File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    answer = 0
    ls = set()
    
    for i in range(1, len(numbers) + 1):
        for j in permutations(numbers, i):
            num = int("".join(j))
            ls.add(num)

    for i in ls:
        if i < 2:
            continue
        for j in range(2, int(i**0.5) + 1):
            if i % j == 0:
                break
        else:
            answer += 1

    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function dfs(now, rest, ls) {
    if (now.length > 0) ls.add(parseInt(now));

    for (let i = 0; i < rest.length; i++) {
        dfs(now + rest[i], rest.slice(0, i) + rest.slice(i + 1), ls);
    }
}

function solution(numbers) {
    var answer = 0;
    let ls = new Set();

    dfs("", numbers, ls);

    ls.forEach(num => {
        if (num < 2) return;
        let tmp = true;
        for (let j = 2; j * j <= num; j++) {
            if (num % j === 0) {
                tmp = false;
                break;
            }
        }
        if (tmp) answer++;
    });

    return answer;
}
